[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70475
* #70375
* #70374



Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

Commit:
[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

priyaramani commented on torch/csrc/jit/mobile/nnc/aot_compiler.cpp:
```diff
@@ -313,9 +314,26 @@ c10::IValue preprocess(
+      LLVMTargetAttrs() = "+neon,+thumb2";
+    } else if (arch == "") {
+      // Do nothing
+    } else {
```
Can we add x86 (aka x86-32) to the list too?

priyaramani commented on torch/csrc/jit/mobile/nnc/aot_compiler.cpp:
```diff
@@ -313,9 +314,26 @@ c10::IValue preprocess(
+    } else if (arch == "aarch64") {
+      LLVMTargetTriple() = "aarch64-unknown-unknown";
+      LLVMTargetAttrs() = "+neon,+thumb2";
+    } else if (arch == "arm") {
```
I guess this is for all armv7, armv8 etc, right?

Commit:
Update on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

Commit:
Update on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

Commit:
Update on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

Commit:
Update on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

Commit:
clang-tidy on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

Commit:
clang-tidy on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

Commit:
Update on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

Commit:
Update on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

Commit:
Update on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

Commit:
Update on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

Commit:
Update on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).

Commit:
Update on "[WIP][TensorExpr] AOTCompiler: specify arch in commandline and compile to asm directly."




Differential Revision: [D33344112](https://our.internmc.facebook.com/intern/diff/D33344112)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33344112).
